### PR TITLE
Transition from using setup.py to pyroject.toml to specify project metadata

### LINKED ISSUE #1801: Transition from using setup.py to pyroject.toml to specify project metadata
### Problem Description
- We currently use **setup.py** and **setup.cfg** to specify the project metadata. This includes requirements which are scatter in `apt.txt`, `requirements.txt`, and `tox.ini`.
- The Python community selected pyproject.toml as a standard way of specifying project metadata, removing the need for all these files. All the project metadata should go in this file. 
- Other modern packaging and dependency management projects, like [Python Poetry](https://python-poetry.org/), only support the more declarative way of using a pyproject.toml file.

### Expected behavior
- Use `pyproject.toml` as the one file for managing project metadata

### PR DESCRIPTION
**Pull request by R-Palazzo**
CU-86azdvakf
Resolve #1801

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86azdvakf SDV - Transition from using setup.py to pyroject.toml to specify project metadata #1801](https://app.clickup.com/t/86azdvakf)

**Comment by sdv-team** [comment]
Task linked: [CU-86azdvakf SDV - Transition from using setup.py to pyroject.toml to specify project metadata #1801](https://app.clickup.com/t/86azdvakf)

File: pyproject.toml [hunk 1]
```diff
@@ -30,23 +41,29 @@
     'sdmetrics>=0.13.0,<0.14',
 ]
 
-pomegranate_requires = [
-    "pomegranate>=0.14.3,<0.15",
-]
+[project.urls]
+"Source Code"= "https://github.com/sdv-dev/SDV/"
+"Issue Tracker" = "https://github.com/sdv-dev/SDV/issues"
```
Comment (gsheni) [inline pyproject.toml:46]:
```suggestion
"Issue Tracker" = "https://github.com/sdv-dev/SDV/issues"
"Changes" = "https://github.com/sdv-dev/SDV/blob/main/HISTORY.md"```

**Review by gsheni — APPROVED** [review]

**Review by amontanez24 — APPROVED** [review]
LGTM!

File: tests/test_tasks.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+"""Tests for the ``tasks.py`` file."""
+from tasks import _get_minimum_versions
+
+
+def test_get_minimum_versions():
+    """Test the ``_get_minimum_versions`` method.
+
+    The method should return the minimum versions of the dependencies for the given python version.
+    If a library is linked to an URL, the minimum version should be the URL.
```
Comment (fealho) [inline tests/test_tasks.py:9]: `a` URL instead of `an` URL

**Review by fealho — APPROVED** [review]

**Pull request merged**